python/pyspark/sql/dataframe.py

```diff
@@ -714,30 +714,45 @@ def count(self):
         [Row(age=2, count=1), Row(age=5, count=1)]
         """
 
-    @dfapi
-    def mean(self):
+    def mean(self, *cols):
```

Comment:
There are lots of duplicated codes here, you could update the `dfapi` to take any number of cols as arguments, then have a different implementation for `count`.